### DISCUSSION TITLE
2022-45: 使用 sccache 加快 Rust 编译速度

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#388] [Announcements]
# 2022-45: 使用 sccache 加快 Rust 编译速度

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/reports/2022-45/

**Comment by Endle** [comment]
感谢分享

不过，折腾这么久感觉收益不是很大啊，4m38s vs 6m20s 。还是说，因为matrix很大，所以这种30%的提升很显著了？

另外
> Databend 的 dev 构建跑在 AWS 的 Self-hosted Runner 上，而且本身就已经配置好了内网的 S3 bucket

带来的额外存储成本高吗？缓存失效的策略需要特别配置吗？

**Reply by Xuanwo** [reply]
> 折腾这么久感觉收益不是很大啊

确实，不过后续的持续运行成本并不是很高，勉强能接受吧

> 因为matrix很大，所以这种30%的提升很显著了？

Databend 每个 PR 要编译 4 次，每次能优化 30% 的时候感觉还可以，不过也不是每次都能用上。目前留着是想后面看看能不能 cache 更多 crate。

> 带来的额外存储成本高吗？缓存失效的策略需要特别配置吗？

不高，s3 很便宜，我们统一配置了保存 7 天的生命周期策略～

**Comment by gaocegege** [comment]
我们现在一次 test 15 min 也没打算去优化 Orz

**Comment by Endle** [comment]
你的 cccache 的改动合并到 sccache 以后，这篇文章需要更新一下吗
